lib/Dialect/FIRRTL/Transforms/IMConstProp.cpp

```diff
@@ -169,41 +264,109 @@ struct IMConstPropPass : public IMConstPropBase<IMConstPropPass> {
     return executableBlocks.count(block);
   }
 
-  bool isOverdefined(Value value) const {
-    auto it = latticeValues.find(value);
+  bool isOverdefined(ValueAndLeafIndex value) const {
+    auto it = find(value);
     return it != latticeValues.end() && it->second.isOverdefined();
   }
 
-  /// Mark the given value as overdefined. This means that we cannot refine a
-  /// specific constant for this value.
-  void markOverdefined(Value value) {
+  // Value is regarded as overdefined if one of lattice values associated with
+  // the value is overdefined.
+  bool isOverdefined(Value value) const {
+    auto [root, index, size] =
```

Comment:
Is this a C++'20 thing?  We use C++'17, but I haven't seen us use this extension before.

Reply:
Yes, I prefer structured bindings over `std::tie` :)

Reply:
We might have had problems with this feature in windows.  You should check MSVC.

Reply:
Thanks, I ran windows ci but it passed so I assume it's fine.